model/collection.py

```diff
@@ -715,6 +725,37 @@ def restrict_to_ready_deliverable_works(
             )
         return query
 
+    def delete(self):
+        """Delete a collection.
+
+        Collections can have hundreds of thousands of
+        LicensePools. This deletes a collection gradually in a way
+        that can be confined to the background and survive interruption.
+        """
+        _db = Session.object_session(self)
+        if not self.marked_for_deletion:
+            raise Exception(
+                "Cannot delete %s: it is not marked for deletion." % self.name
+            )
+
+        # Delete all the license pools. This should be the only part
+        # of the application where LicensePools are permanently
+        # deleted.
+        for i, pool in enumerate(self.licensepools):
+            _db.delete(pool)
+            if not i % 100:
+                _db.commit()
+            # TODO: If the LicensePool's Work has no other
+            # LicensePools, the Work should be removed from the search
+            # index. We can't do this now because we currently have no
+            # way of removing something from the search index. This
+            # only affects search engine performance -- it won't cause
+            # phantom books to show up in search results.
```

Comment:
There is a remove_work method in the search code. Probably won't make much difference but I don't think it would be that hard to do this.